src/models/acl.ts

```diff
@@ -0,0 +1,64 @@
+import * as yml from 'js-yaml';
+import { Validator as JSONValidator } from 'jsonschema';
+
+import OwnerAcl from './acl/owner';
+import ReleaseOwnerAcl from './acl/release-owner';
+import AclBase, { IAclBase } from './acl/base';
+
+const validator = new JSONValidator();
+
+const ACL_DATA_SCHEMA = {
+  id: '/AclFile',
+  type: 'object',
+  properties: {
+    whitelist: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    paths: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    description: { type: 'string' },
+    block_message: { type: 'string' },
+    groups: { items: { type: 'string' }, type: 'array' },
+    exclude_paths: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+  },
+  required: ['paths'],
+};
+
+function parseYaml(
+  yamlText: string,
+): IAclBase & ({ owners?: string[]; release_owners?: string[] }) {
```

Comment:
Can you hoist out this union into it's own `type`

Reply:
Addressed in #26